srtcore/buffer.cpp

```diff
@@ -576,13 +577,34 @@ int CSndBuffer::readData(const int offset, CPacket& w_packet, steady_clock::time
     // TODO: FR #930. Use source time if it is provided.
     w_srctime = getSourceTime(*p);
 
+    // This function is called when packet retransmission is triggered.
+    // Therefore we are setting the rexmit time.
+    p->m_tsRexmitTime = steady_clock::now();
+
     HLOGC(dlog.Debug,
           log << CONID() << "CSndBuffer: getting packet %" << p->m_iSeqNo << " as per %" << w_packet.m_iSeqNo
               << " size=" << readlen << " to send [REXMIT]");
 
     return readlen;
 }
 
+int CSndBuffer::getPacketTime(const int offset, srt::sync::steady_clock::time_point& tsLastRextim)
```

Comment:
`tsLastRextim` -> `w_tsLastRexmit`

Additionally, you didn't initialize the `tsLastRextim` ref with zero time, while you rely on it when calling. I think it would be simply better if you return zero time or `p_m_tsRexmitTime` instead of returning -1 or 0 and passing by reference, since your calls exactly rely on it.